core/lib/makeswift/components/site-header/site-header.client.tsx

```diff
@@ -0,0 +1,101 @@
+'use client';
+
+import {
+  type ComponentPropsWithoutRef,
+  createContext,
+  forwardRef,
+  type PropsWithChildren,
+  type ReactNode,
+  type Ref,
+  useContext,
+} from 'react';
+
+import { useStreamable } from '@/vibes/soul/lib/streamable';
+import { HeaderSection } from '@/vibes/soul/sections/header-section';
+
+type Props = ComponentPropsWithoutRef<typeof HeaderSection>;
+
+const PropsContext = createContext<Props>({
+  navigation: {
+    accountHref: '',
+    cartHref: '',
+    searchHref: '',
+    links: [],
+  },
+});
+
+export const PropsContextProvider = ({ value, children }: PropsWithChildren<{ value: Props }>) => (
+  <PropsContext.Provider value={value}>{children}</PropsContext.Provider>
+);
+
+export const MakeswiftHeader = forwardRef(
+  (
+    {
+      banner,
+      links,
+      logo,
+      linksPosition,
+    }: {
+      banner?: {
+        show: boolean;
+        allowClose: boolean;
+        children?: ReactNode;
+      };
+      links: Array<{
+        text: string;
+        link: { href: string };
+        subLinks: Array<{ text: string; link: { href: string } }>;
+      }>;
+      logo?: {
+        desktopSrc?: string;
+        desktopAlt: string;
+        desktopWidth?: number;
+        mobileSrc?: string;
+        mobileAlt: string;
+        mobileWidth?: number;
+        link?: { href: string };
+      };
+      linksPosition: 'center' | 'left' | 'right';
+    },
+    ref: Ref<HTMLDivElement>,
+  ) => {
+    const { navigation: passedProps, banner: passedBanner } = useContext(PropsContext);
+    const bannerElement = banner?.children ? { children: banner.children } : passedBanner;
+    const passedLinks = useStreamable(passedProps.links);
```

Comment:
This component will suspend if we use `useStreamable`. We should handle this by adding a `Suspense` boundary in the parent component and rendering a skeleton as a fallback